apps/src/calc/calc.js

```diff
@@ -805,16 +817,11 @@ function tokenListForEvaluation_(userSet, targetSet) {
   var evaluation = userSet.evaluate();
 
   // Check for div zero
-  var divZeroInUserSet = false;
   if (evaluation.err) {
-    if (evaluation.err instanceof ExpressionNode.DivideByZeroError) {
-      divZeroInUserSet = true;
-    } else {
-      throw evaluation.err;
+    if (evaluation.err instanceof ExpressionNode.DivideByZeroError &&
+        !utils.isInfiniteRecursionError(evaluation.err)) {
+      console.log('Unexpected error: ' + evaluation.err);
```

Comment:
Nope. The logic is wrong. What it should be saying is if this isn't a DivideByZeroError and it isn't infinite recursion (two expected errors), log our unexpected error. Nice catch.